pkg/k2/constructs/constructs.go

```diff
@@ -1,70 +1,44 @@
 package constructs
 
 import (
-	"reflect"
-	"text/template"
-
 	"github.com/klothoplatform/klotho/pkg/construct"
+	"github.com/klothoplatform/klotho/pkg/k2/constructs/template"
+	"github.com/klothoplatform/klotho/pkg/k2/constructs/template/property"
 	"github.com/klothoplatform/klotho/pkg/k2/model"
-	"github.com/klothoplatform/klotho/pkg/k2/reflectutil"
 )
 
 type (
 	ResourceOwner interface {
 		GetResource(resourceId string) (resource *Resource, ok bool)
 		SetResource(resourceId string, resource *Resource)
 		GetResources() map[string]*Resource
-		GetTemplateResourcesIterator() Iterator[string, ResourceTemplate]
-		InterpolationSource
+		GetTemplateResourcesIterator() template.Iterator[string, template.ResourceTemplate]
```

Comment:
TODO (for later, not this PR): https://go.dev/wiki/RangefuncExperiment ([1.23](https://tip.golang.org/doc/go1.23#language) adds it to the language)